archivessnake/create_daos.py

```diff
@@ -0,0 +1,205 @@
+
+    Args:
+        dip_info (str): path to pickle file
+
+    Returns:
+        DIP information (array)
+    """
+    dip_data = pickle.load(open(dip_file, "rb"))
+    return dip_data
+
+
+def get_uris(aspace, resource, series):
+    """Takes a resource and a series from that resource to return all file-level descendants of the series that do not have linked digital objects.
+
+    Args:
+        aspace (asnake.aspace.ASpace): instantiation of ASpace
+        resource (str): ArchivesSpace ID of resource. Example: 123
+        series (str): ArchivesSpace ID of series. Example: 4321
+
+    Returns:
+        list of ArchivesSpace URIs (array)
+    """
+    lsrm_tree = aspace.client.get(
+        "repositories/2/resources/{}/tree".format(resource)).json()
+    lsrm_tree.get('children')[1].get('children')
+    series_tree = [child for child in lsrm_tree.get(
+        'children') if child.get("id") == series][0]
+    list_of_lsrm_uris = []
+    for subseries in series_tree.get("children"):
+        for file_level in subseries.get("children"):
+            if "digital_object" not in file_level.get("instance_types"):
+                list_of_lsrm_uris.append(file_level.get('record_uri'))
+    return list_of_lsrm_uris
+
+
+def match_dip_to_aip(aspace, ao_uri, dip_list):
+    """Using box and folder indicators, matches an ArchivesSpace component to a DIP.
+
+    Args:
+        aspace (asnake.aspace.ASpace): instantiation of ASpace
+        ao_uri (str): ArchivesSpace URI of an archival object
+        dip_list (array): List of dictionaries of DIP information
+
+    Returns:
+        DIP information (dictionary)
+    """
+    ao = aspace.client.get(
+        ao_uri, params={"resolve": ["top_container"]}).json()
+    box = ao.get('instances')[0].get('sub_container').get(
+        'top_container').get('_resolved').get('indicator')
+    folder = ao.get('instances')[0].get('sub_container').get('indicator_2')
+    dips_same_box = [dip for dip in dip_list if dip.get("box") == box]
+    dip = [dip for dip in dips_same_box if dip.get("folder") == folder]
+    if len(dip) == 1:
+        dip = dip[0]
+        return dip
+    else:
+        raise Exception(
+            "Error for {}: {} matches found, expected 1.".format(
+                ao_uri, len(dip)))
+
+
+def get_dip_info(dip):
+    """Parses DIP information to return DIP UUID, title to be used as DAO title, and UUID of associated AIP.
+
+    Args:
+        dip (dictionary): DIP information
+
+    Returns:
+        UUID, file title, and associated AIP UUID (tuple)
+    """
+    dip_uuid = dip.get("uuid")
+    title = dip.get("current_full_path")[89:-37]
+    aip_uuid = dip.get("related_packages")[0][13:-1]
+    return dip_uuid, title, aip_uuid
+
+
+def post_dao(dao_url, aspace, dip_uuid, title, aip_uuid):
+    """Creates a digital object in ArchivesSpace
+
+    Args:
+        dao_url (str): URL for posting ArchivesSpace digital objects
```

Comment:
There's a related comment above, but I wonder if it's necessary to pass the `dao_url` as an argument if it's only getting used within the scope of this function?